phoenix-core/src/it/java/org/apache/phoenix/end2end/ArithmeticQueryIT.java

```diff
@@ -985,4 +985,64 @@ public void testFloatingPointMultiplicationUpsert() throws Exception {
         assertTrue(rs.next());
         assertEquals(-1.0f, rs.getFloat(1), 0.001);
     }
+    
+    @Test
+    public void testSystemTableHasDoubleForExponentialNumber() throws Exception {
```

Comment:
@JamesRTaylor  here is a test that verifies that the internal representation of an exponential number is infact a Double.